docs/tutorials/text_prediction/customization.md

```diff
@@ -1,274 +1,204 @@
-Here we specify **bayesopt** as the searcher.
+The pre-registered configurations provide a bunch of good default hyperparameters. A common workflow is to first train a model with one of the presets and then tune part of hyperparameters to see if the performance can be better. The following is an example about how to do this in AutoGluon Text. You can directly add/changes keys in the hyperparameter dictionary. In the example, we change the number of training epochs to 5.
 
 
 ```{.python .input}
-hyperparameters['hpo_params'] = {
-    'search_strategy': 'bayesopt'
-}
-
-predictor_mrpc_bo = task.fit(train_data, label='label',
-                                hyperparameters=hyperparameters,
-                                time_limits=60 * 2,
-                                num_trials=2,  # increase this to get good performance in your applications
-                                ngpus_per_trial=1, seed=123,
-                                output_directory='./ag_mrpc_custom_space_fifo_bo')
+hyperparameters = ag_text_presets.create('electra_small_fuse_late')
+hyperparameters['models']['MultimodalTextModel']['search_space']['optimization.num_train_epochs'] = 5
+
+predictor.fit(train_data, hyperparameters=hyperparameters, time_limit=30, seed=123)
 ```
 
+### Register Your Own Configuration
 
-```{.python .input}
-dev_score = predictor_mrpc_bo.evaluate(dev_data, metrics=['acc', 'f1'])
-print('Best Config = {}'.format(predictor_mrpc_bo.results['best_config']))
-print('Total Time = {}s'.format(predictor_mrpc_bo.results['total_time']))
-print('Accuracy = {:.2f}%'.format(dev_score['acc'] * 100))
-print('F1 = {:.2f}%'.format(dev_score['f1'] * 100))
-```
+You can also register the hyperparameters to `ag_text_presets`.
 
 
 ```{.python .input}
-predictions = predictor_mrpc_bo.predict(dev_data)
-prediction1 = predictor_mrpc_bo.predict({'sentence1': [sentence1], 'sentence2': [sentence2]})
-prediction1_prob = predictor_mrpc_bo.predict_proba({'sentence1': [sentence1], 'sentence2': [sentence2]})
-print('A = "{}"'.format(sentence1))
-print('B = "{}"'.format(sentence2))
-print('Prediction = "{}"'.format(prediction1[0] == 1))
-print('Prob = "{}"'.format(prediction1_prob[0]))
-print('')
-prediction2 = predictor_mrpc_bo.predict({'sentence1': [sentence1], 'sentence2': [sentence3]})
-prediction2_prob = predictor_mrpc_bo.predict_proba({'sentence1': [sentence1], 'sentence2': [sentence3]})
-print('A = "{}"'.format(sentence1))
-print('B = "{}"'.format(sentence3))
-print('Prediction = "{}"'.format(prediction2[0] == 1))
-print('Prob = "{}"'.format(prediction2_prob[0]))
+@ag_text_presets.register()
+def electra_small_fuse_late_train5():
+    hyperparameters = ag_text_presets.create('electra_small_fuse_late')
+    hyperparameters['models']['MultimodalTextModel']['search_space']['optimization.num_train_epochs'] = 5
+    return hyperparameters
+
+predictor.fit(train_data, presets='electra_small_fuse_late_train5', time_limit=60, seed=123)
```

Comment:
This seems like a nice UX. 
@Innixma @gradientsky  Should we add this to core so we can use in TabularPredictor as well?

Reply:
It is quite reasonable, but probably won't make it for v0.1 unless we have spare time or someone wants to take it up.

Also, not sure how much code is in this register logic, we'd need to extract it from gluonnlp I think.